git_flow.rst

```diff
@@ -191,6 +191,56 @@ This synchronizes their local ``master`` branch with the ``master`` branch of
 the ``upstream`` remote repo, and deletes their local ``fix_branch`` (which is
 no longer needed, since its changes have been merged into the upstream master).
 
+Debugging with Git
+
+Occasionally as a developer you might run into a bug that you are uncertain when it was introduced. Instead of going through all the commits manually that have occurred since the code last worked, you can use the git bisect command::
+
+    $ git bisect
+
+What it does:
+This command is a way of quickly finding where a problem was introduced in a particular commit. It does so by performing a “binary” search between two different states of the code and via process of elimination - where the issue was first introduced. 
+
+How to use it:
+First, you will need to find a commit where the issue is not present (where it is working). After doing a search through the git log from the time you know things were working, confirm that this commit  does indeed work by switching your checkout to it instead::
+
+    $ git checkout abc1234
+
+Then you will need to find a commit where things are not working. You will do this by getting the current commit hash.:
```

Comment:
This needs to be `::` to render properly (check "view" button above to see how it looks currently)